test/new-e2e/tests/agent-metric-logs/linux-log-agent/file_tailing_utils_test.go

```diff
@@ -0,0 +1,178 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	componentos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/stretchr/testify/assert"
+)
+
+// generateLog generates x content with y reccurrence and verifies log contents.
+func generateLog(s *LinuxVMFakeintakeSuite, content string, recurrence int) {
+	// Determine the OS and set the appropriate log path and command.
+	var logPath, cmd, checkCmd string
+	t := s.T()
+	t.Helper()
+
+	osType := s.Env().VM.GetOSType()
+	var os string
+
+	switch osType {
+	case componentos.WindowsType:
+		os = "windows"
+		t.Log("Generating Windows log.")
+		logPath = "C:\\logs\\hello-world.log"
+		cmd = fmt.Sprintf("echo %s > %s", strings.Repeat(content+" ", recurrence), logPath)
+		checkCmd = fmt.Sprintf("Get-Content %s", logPath)
+	default: // Assuming Linux if not Windows.
+		os = "linux"
+		t.Log("Generating Linux log.")
+		logPath = "/var/log/hello-world.log"
+		cmd = fmt.Sprintf("echo '%s' | sudo tee -a %s", strings.Repeat(content+" ", recurrence), logPath)
+		checkCmd = fmt.Sprintf("sudo cat %s", logPath)
+	}
+
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		// Generate the log content
+		output, err := s.Env().VM.ExecuteWithError(cmd)
+		if err != nil {
+			assert.FailNowf(t, "Having issue generating %s log with error: %s", os, output)
+		}
+		// Verify the log content locally
+		output, err = s.Env().VM.ExecuteWithError(checkCmd)
+		if err != nil {
+			assert.FailNowf(t, "Log content %s not found, instead received:: %s", content, output)
+		}
+		if strings.Contains(output, content) {
```

Comment:
This check will give incorrectly pass if you ask for the same content multiple times. The first content addition may succeed, and the second one may fail, but if the second one fails, this check will pass because it was added in the first call to `generateLogs`.

This is also not a blocking comment for this review, just pointing this out for future improvement.